DEVELOPMENT.md

```diff
@@ -0,0 +1,107 @@
+
+*Note that this requires macOS 10.15.2+ as a build-time dependency. The resulting builds should work on versions as old as macOS 10.10*
+
+2. Install [Homebrew](https://brew.sh/)
+
+**Linux**
+ 1. Create an Ubuntu 20.04 environment:
+    * The `ubuntu:20.04` docker container is a good option and this is the approach the CI build uses
+    * A virtual machine or even WSL on Windows will work too
+    * Running on bare metal is of course fine too, but other versions of Ubuntu are untested and unsupported.
+
+*Note that Ubuntu 20.04 is only a build-time dependency. The resulting builds are intended to run on any Linux distro*
+
+ 2. Make sure the `sudo` package is installed (it is not installed by default in the docker image)
+
+## Running a Build
+
+**WARNING**: The scripts will attempt to automatically install build dependencies by default! If you don't want this you can disable it first with `export INSTALL_DEPS=0`.
+
+The details of the required dependencies will not be documented here, so it is recommended to just let the scripts handle it - you can see what will be installed in `scripts/install_dependencies.sh`
+
+Build:
+
+```
+bash build.sh <arch>
+```
+
+Clean:
+
+```
+bash clean.sh <arch>
+```
+
+Current architectures:
+* linux_x86_64
+* windows_amd64
+* darwin
+
+Final packages will be deployed in the `./_packages/build_<arch>/` directory.
+
+By default the scripts will not build nextpnr-ecp5. See [this section](#building-nextpnr-ecp5) for details on how to build this too.
+
+## Disabling parts of the build
+
+The build scripts define many variables that may be used to disable parts of the build during development - see `build.sh` for details.
+
+If you place a `.env` file in the root of the repo then the bash scripts will source it. You can use this as a convenient way to locally override these variables without accidentally committing changes (the file is already in `.gitignore`).
+
+## Building nextpnr-ecp5
+
+These scripts currently require an Ubuntu 20.04 environment ([as specified here](#prerequisites)) to generate the ECP5 device databases. The device databases are represented as a text-based set of instructions for the nextpnr Binary Blob Assembler (a "BBA file"). It is important that the same nextpnr and libtrellis git commits are used for the whole build to avoid the BBA files getting out of sync with the compiled code.
+
+In the Ubuntu 20.04 environment, run:
+
+`./build_bba.sh`
+
+This will result in a package generated at:
+
+`./_packages/build_linux_x86_64/ecp5-bba-noarch-nightly.tar.gz`
+
+If you are building in the same working folder, you can then simply run:
+
+`COMPILE_NEXTPNR_ECP5=1 ./build.sh <arch>`
+
+If you are building the bba files in a separate working folder, you will need to copy the bba package into `./_packages/build_linux_x86_64/ecp5-bba-noarch-nightly.tar.gz` before running `build.sh` (this is how the CI build works)
```

Comment:
They might have to `mkdir -p` that directory first, if it doesn't exist yet in a clean tree. 
I placed that file in the repository root before and that seemed to work as well.

Reply:
Edited to explicitly mention that the folder needs to be created (the artifact download in the CI job implicitly does it for us)